src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -352,6 +352,23 @@ public int compare(HaxeNamedComponent o1, HaxeNamedComponent o2) {
     return result;
   }
 
+  public static List<HaxeNamedComponent> getAllNamedSubComponentsFromClassType(HaxeClass haxeClass, HaxeComponentType... fromTypes) {
+    List<HaxeNamedComponent> components = getNamedSubComponents(haxeClass);
+    List<HaxeComponentType> types = Arrays.asList(fromTypes);
+    HaxeComponentType type = HaxeComponentType.typeOf(haxeClass);
+    if (types.contains(type)) {
+      components.addAll(getNamedSubComponents(haxeClass));
+    }
+    PsiClass[] supers = haxeClass.getSupers();
+    for (PsiClass superComponent : supers) {
+      type = HaxeComponentType.typeOf(superComponent);
+      if (types.contains(type)) {
+        components.addAll(getNamedSubComponents((HaxeClass)superComponent));
+      }
+    }
```

Comment:
Rather than duplicating this loop (which only gets the second level in the hierarchy), wouldn't it be better to go recursive?

Reply:
I guess i was expecting `getSupers()` to give me a list of all supers not just the first level, but since its only level at the time recursive makes more sense :)